### PR TITLE
Add a package which builds SVG sprites

### PR DESCRIPTION
**Pull request by thure**
Packages which depend directly on `@fluentui/react-icons` can cause very lengthy build times, since some bundler configurations will cause the bundler to iterate through each of the 8,833 icons in this project. This ~is~ was true at least for Storybook consuming `@fluentui/react-components`.

The situation calls for a more comprehensive solution than even the one contained in this PR. I think the ideal solution for a React stack would be to offer a module which tracks which icons are used by an app (maybe similar to the `makeStyles`/`useStyles` pattern), and render either a one-off sprite during development or a sprite optimized for exactly the icons used in the app in production.

As a first step, though, the spriting solution in this PR offers these benefits:
- Developers don’t need to install this package as a dependency
- The user agent can cache requests for sprites
- Icon assets no longer affect project bundle size

SVG sprites have been well supported by user agents [for more than 5 years by now](https://caniuse.com/mdn-svg_elements_symbol). This package uses [`svgstore`](https://github.com/svgstore/svgstore) to generate the sprites.

### CODE REVIEW

File: packages/svg-sprites/svgo.config.yaml
Comment (bheston) [inline packages/svg-sprites/svgo.config.yaml:2]: Can `precision` go in here as well?

File: packages/svg-sprites/unfill.js
Comment (bheston) [inline packages/svg-sprites/unfill.js:1]: Can we have a unified treatment for icons, at least within svg type? If I recall the `none` fill comes from the root `svg` element, which is beneficial to remove for size in all cases. The `212121` fill comes from the `path`. In the original package we opted to specify no fill to remove any _opinion_ on color and allow consumers to (likely) set to `currentcolor` in their style sheets. I suggest going forward with that since it would work in this new package, and would also make the download size smaller. I do agree that `currentcolor` is probably very common, but since it's not a given I would lean toward being more concise.

Can we share the file from the other package or move it to a utilities directory?

**Review by bheston — CHANGES_REQUESTED** [review]
I wasn't familiar with that. Nice optimization for those who could use it.

File: packages/svg-sprites/svgo.config.yaml
Comment (thure) [reply packages/svg-sprites/svgo.config.yaml:2]: I believe so, yes!

File: packages/svg-sprites/unfill.js
Comment (thure) [reply packages/svg-sprites/unfill.js:1]: I'll look into that, thanks for the feedback.

File: packages/svg-sprites/svgo.config.yaml
Comment (thure) [reply packages/svg-sprites/svgo.config.yaml:2]: Is there a standard value for `precision` we should use here?

**Comment by thure** [comment]
@bheston I've given this another stab informed by your feedback, and I've published and tested the result. As far as I can tell, everything works as expected, and now precision is correct and there're no superfluous `fill` attributes ✨

**Comment by thure** [comment]
I am also considering a follow-up where this package can be added as a developer dependency, then used to select specific icons to add to a single sprite for a project to use.

A comparison with that idea and this PR is here: https://github.com/thure/fluentui-system-icons/pull/1/files

File: packages/svg-sprites/package.json
Comment (bheston) [inline packages/svg-sprites/package.json:2]: Probably rename the folder for consistency.

**Review by bheston — APPROVED** [review]

File: packages/svg-sprites/package.json
Comment (tomi-msft) [reply packages/svg-sprites/package.json:2]: Maybe `"@fluentui/react-svg-icon-sprites"`

Comment (thure) [reply packages/svg-sprites/package.json:2]: 'fluent' seems redundant since it's already scoped to @fluentui, and the adjacent package is named `@fluentui/svg-icons`.

Maybe `@fluentui/svg-sprites`? I'll go with that name and adjust the directory name for now, let me know if it should change.

Comment (thure) [reply packages/svg-sprites/package.json:2]: Ah, and that happens to be the name of the directory already.

**Comment by thure** [comment]
Looks like it just needs a review from someone with write access… @bheston last time your approval wasn't enough to merge.

File: packages/svg-sprites/README.md
Comment (spencer-nelson) [inline packages/svg-sprites/README.md:18]: Does this reference need to change with the package name change?

Comment (thure) [reply packages/svg-sprites/README.md:18]: Ah, indeed, thanks for catching this.

**Comment by vnbaaij** [comment]
As this is enabling to work with sprites, I was wondering if it would make sense to 'pack' all the variants of an icon in 1 sprite. The size of the individual icons is so small that even a packed version is still a very small download (largest one is ~20k).
The way this can be achieved in `package.json` is to replace the `sprites` script with the following :
```
"sprites": "mkdir -p ./sprites && for j in ./icons/**; do svgstore  -o \"./sprites/$(basename $j)/$(basename $j).sprite.svg\" \"$j/*.svg\"; done",
```

File: packages/svg-sprites/package.json
Comment (tomi-msft) [inline packages/svg-sprites/package.json:3]: Update version to 1.1.167 to be in line with other icon packages

File: packages/svg-sprites/package.json
Comment (bheston) [reply packages/svg-sprites/package.json]: I was suggesting the package and directory names should both be `svg-icon-sprites`. By itself I don't think `svg-sprites` alone sufficiently identifies them as icons. It could be any svg content, but I think the consistency in naming between all modern `icon` packages is beneficial.

**Comment by bheston** [comment]
> Looks like it just needs a review from someone with write access… @bheston last time your approval wasn't enough to merge.

I believe @spencer-nelson was able to push through one of my previous PRs.

**Review by tomi-msft — APPROVED** [review]

**Pull request merged**
